### ISSUE TITLE
Missing character warnings when setting samples to certain values

### ISSUE THREAD

**Issue by leingang** [pgf-tikz/pgfplots#456]
Here is a MWE:

```TeX
\documentclass{standalone}
\usepackage{pgfplots}
\begin{document}
\begin{tikzpicture}
    \begin{axis}
        \addplot[samples=51] {x^2};
    \end{axis}
\end{tikzpicture}
\end{document}
```

The document produces the expected output, but there are some warnings in the log file:

```
Missing character: There is no . in font nullfont!
Missing character: There is no 0 in font nullfont!
Missing character: There is no 1 in font nullfont!
Missing character: There is no p in font nullfont!
Missing character: There is no t in font nullfont!
```

I know that these warnings come up when non-drawing tokens are inside the `tikzpicture` environment. The tokens `.01pt` looks like some length that's being read as text instead of as a length.  Curiously, the messages are not present if I set `samples=51` instead, which makes me wonder if it's happening around x=0.

**Comment by hmenke** [comment]
Sorry, I can't reproduce this one. Here is my log file (with an additional `\listfiles` before `\documentclass`).

<details>

<summary>Log file</summary>

```
This is pdfTeX, Version 3.141592653-2.6-1.40.24 (TeX Live 2022) (preloaded format=pdflatex)
 restricted \write18 enabled.
entering extended mode
(./test.tex
LaTeX2e <2022-11-01> patch level 1
L3 programming layer <2023-02-07>
(/opt/texlive/2022/texmf-dist/tex/latex/standalone/standalone.cls
Document Class: standalone 2022/10/10 v1.3b Class to compile TeX sub-files stan
dalone
(/opt/texlive/2022/texmf-dist/tex/latex/tools/shellesc.sty)
(/opt/texlive/2022/texmf-dist/tex/generic/iftex/ifluatex.sty
(/opt/texlive/2022/texmf-dist/tex/generic/iftex/iftex.sty))
(/opt/texlive/2022/texmf-dist/tex/latex/xkeyval/xkeyval.sty
(/opt/texlive/2022/texmf-dist/tex/generic/xkeyval/xkeyval.tex
(/opt/texlive/2022/texmf-dist/tex/generic/xkeyval/xkvutils.tex
(/opt/texlive/2022/texmf-dist/tex/generic/xkeyval/keyval.tex))))
(/opt/texlive/2022/texmf-dist/tex/latex/standalone/standalone.cfg)
(/opt/texlive/2022/texmf-dist/tex/latex/base/article.cls
Document Class: article 2022/07/02 v1.4n Standard LaTeX document class
(/opt/texlive/2022/texmf-dist/tex/latex/base/size10.clo)))
(/opt/texlive/2022/texmf-dist/tex/latex/pgfplots/pgfplots.sty
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplots.revision.tex)
(/opt/texlive/2022/texmf-dist/tex/latex/graphics/graphicx.sty
(/opt/texlive/2022/texmf-dist/tex/latex/graphics/graphics.sty
(/opt/texlive/2022/texmf-dist/tex/latex/graphics/trig.sty)
(/opt/texlive/2022/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)
(/opt/texlive/2022/texmf-dist/tex/latex/graphics-def/pdftex.def)))
(/opt/texlive/2022/texmf-dist/tex/latex/pgf/frontendlayer/tikz.sty
(/opt/texlive/2022/texmf-dist/tex/latex/pgf/basiclayer/pgf.sty
(/opt/texlive/2022/texmf-dist/tex/latex/pgf/utilities/pgfrcs.sty
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/utilities/pgfutil-common.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/utilities/pgfutil-latex.def)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/utilities/pgfrcs.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/pgf.revision.tex)))
(/opt/texlive/2022/texmf-dist/tex/latex/pgf/basiclayer/pgfcore.sty
(/opt/texlive/2022/texmf-dist/tex/latex/pgf/systemlayer/pgfsys.sty
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/systemlayer/pgfsys.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/utilities/pgfkeys.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/utilities/pgfkeyslibraryfiltered.
code.tex)) (/opt/texlive/2022/texmf-dist/tex/generic/pgf/systemlayer/pgf.cfg)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/systemlayer/pgfsys-pdftex.def
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/systemlayer/pgfsys-common-pdf.def
)))
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/systemlayer/pgfsyssoftpath.code.t
ex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/systemlayer/pgfsysprotocol.code.t
ex)) (/opt/texlive/2022/texmf-dist/tex/latex/xcolor/xcolor.sty
(/opt/texlive/2022/texmf-dist/tex/latex/graphics-cfg/color.cfg)
(/opt/texlive/2022/texmf-dist/tex/latex/graphics/mathcolor.ltx))
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcore.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmath.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathutil.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathparser.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.basic.code.
tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.trigonometr
ic.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.random.code
.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.comparison.
code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.base.code.t
ex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.round.code.
tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.misc.code.t
ex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.integerarit
hmetics.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathcalc.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmathfloat.code.tex))
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfint.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcorepoints.code.tex
)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcorepathconstruct.c
ode.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcorepathusage.code.
tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcorescopes.code.tex
)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcoregraphicstate.co
de.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcoretransformations
.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcorequick.code.tex)

(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcoreobjects.code.te
x)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcorepathprocessing.
code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcorearrows.code.tex
)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcoreshade.code.tex)
 (/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcoreimage.code.tex
)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcoreexternal.code.t
ex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcorelayers.code.tex
)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcoretransparency.co
de.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcorepatterns.code.t
ex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/basiclayer/pgfcorerdf.code.tex)))
 (/opt/texlive/2022/texmf-dist/tex/generic/pgf/modules/pgfmoduleshapes.code.tex
) (/opt/texlive/2022/texmf-dist/tex/generic/pgf/modules/pgfmoduleplot.code.tex)

(/opt/texlive/2022/texmf-dist/tex/latex/pgf/compatibility/pgfcomp-version-0-65.
sty)
(/opt/texlive/2022/texmf-dist/tex/latex/pgf/compatibility/pgfcomp-version-1-18.
sty)) (/opt/texlive/2022/texmf-dist/tex/latex/pgf/utilities/pgffor.sty
(/opt/texlive/2022/texmf-dist/tex/latex/pgf/utilities/pgfkeys.sty
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/utilities/pgfkeys.code.tex))
(/opt/texlive/2022/texmf-dist/tex/latex/pgf/math/pgfmath.sty
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/math/pgfmath.code.tex))
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/utilities/pgffor.code.tex))
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/frontendlayer/tikz/tikz.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/libraries/pgflibraryplothandlers.
code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/modules/pgfmodulematrix.code.tex)

(/opt/texlive/2022/texmf-dist/tex/generic/pgf/frontendlayer/tikz/libraries/tikz
librarytopaths.code.tex)))
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplots.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplotscore.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/sys/pgfplotssysgeneric.code.
tex))
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/libs/pgfplotslibrary.code.te
x)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/oldpgfcompatib/pgfplotsoldpg
fsupp_loader.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/libraries/pgflibraryfpu.code.tex)
Package pgfplots: loading complementary utilities for your pgf version...

(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/oldpgfcompatib/pgfplotsoldpg
fsupp_pgfutil-common-lists.tex))
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/util/pgfplotsutil.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/liststructure/pgfplotslistst
ructure.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/liststructure/pgfplotslistst
ructureext.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/liststructure/pgfplotsarray.
code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/liststructure/pgfplotsmatrix
.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/numtable/pgfplotstableshared
.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/liststructure/pgfplotsdeque.
code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/util/pgfplotsbinary.code.tex

(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/util/pgfplotsbinary.data.cod
e.tex))
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/util/pgfplotsutil.verb.code.
tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/libs/pgflibrarypgfplots.surf
shading.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/sys/pgflibrarypgfplots.surfs
hading.pgfsys-pdftex.def)))
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/util/pgfplotscolormap.code.t
ex
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/util/pgfplotscolor.code.tex)
)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplotsstackedplots.code.te
x)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplotsplothandlers.code.te
x
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplotsmeshplothandler.code
.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplotsmeshplotimage.code.t
ex)))
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplots.scaling.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplotscoordprocessing.code
.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplots.errorbars.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplots.markers.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplotsticks.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/pgfplots.paths.code.tex)
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/frontendlayer/tikz/libraries/tikz
librarydecorations.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/modules/pgfmoduledecorations.code
.tex))
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/frontendlayer/tikz/libraries/tikz
librarydecorations.pathmorphing.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/libraries/decorations/pgflibraryd
ecorations.pathmorphing.code.tex))
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/frontendlayer/tikz/libraries/tikz
librarydecorations.pathreplacing.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/libraries/decorations/pgflibraryd
ecorations.pathreplacing.code.tex))
(/opt/texlive/2022/texmf-dist/tex/generic/pgfplots/libs/tikzlibrarypgfplots.con
tourlua.code.tex))
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/frontendlayer/tikz/libraries/tikz
libraryplotmarks.code.tex
(/opt/texlive/2022/texmf-dist/tex/generic/pgf/libraries/pgflibraryplotmarks.cod
e.tex))) (/opt/texlive/2022/texmf-dist/tex/latex/l3backend/l3backend-pdftex.def
) (./test.aux)
(/opt/texlive/2022/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/opt/texlive/2022/texmf-dist/tex/latex/epstopdf-pkg/epstopdf-base.sty
(/opt/texlive/2022/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))

Package pgfplots Warning: running in backwards compatibility mode (unsuitable t
ick labels; missing features). Consider writing \pgfplotsset{compat=1.18} into 
your preamble.
 on input line 4.

[1{/opt/texlive/2022/texmf-var/fonts/map/pdftex/updmap/pdftex.map}] (./test.aux
)

 *File List*
standalone.cls    2022/10/10 v1.3b Class to compile TeX sub-files standalone
shellesc.sty    2019/11/08 v1.0c unified shell escape interface for LaTeX
ifluatex.sty    2019/10/25 v1.5 ifluatex legacy package. Use iftex instead.
   iftex.sty    2022/02/03 v1.0f TeX engine tests
 xkeyval.sty    2022/06/16 v2.9 package option processing (HA)
 xkeyval.tex    2014/12/03 v2.7a key=value parser (HA)
standalone.cfg    2022/10/10 v1.3b Default configuration file for 'standalone' 
class
 article.cls    2022/07/02 v1.4n Standard LaTeX document class
  size10.clo    2022/07/02 v1.4n Standard LaTeX file (size option)
pgfplots.sty    2021/05/15 v1.18.1 Data Visualization (1.18.1)
graphicx.sty    2021/09/16 v1.2d Enhanced LaTeX Graphics (DPC,SPQR)
graphics.sty    2022/03/10 v1.4e Standard LaTeX Graphics (DPC,SPQR)
    trig.sty    2021/08/11 v1.11 sin cos tan (DPC)
graphics.cfg    2016/06/04 v1.11 sample graphics configuration
  pdftex.def    2022/09/22 v1.2b Graphics/color driver for pdftex
    tikz.sty    2023-01-15 v3.1.10 (3.1.10)
     pgf.sty    2023-01-15 v3.1.10 (3.1.10)
  pgfrcs.sty    2023-01-15 v3.1.10 (3.1.10)
pgfrcs.code.tex
 pgfcore.sty    2023-01-15 v3.1.10 (3.1.10)
  pgfsys.sty    2023-01-15 v3.1.10 (3.1.10)
pgfsys.code.tex
pgfsyssoftpath.code.tex    2023-01-15 v3.1.10 (3.1.10)
pgfsysprotocol.code.tex    2023-01-15 v3.1.10 (3.1.10)
  xcolor.sty    2022/06/12 v2.14 LaTeX color extensions (UK)
   color.cfg    2016/01/02 v1.6 sample color configuration
mathcolor.ltx
pgfcore.code.tex
pgfcomp-version-0-65.sty    2023-01-15 v3.1.10 (3.1.10)
pgfcomp-version-1-18.sty    2023-01-15 v3.1.10 (3.1.10)
  pgffor.sty    2023-01-15 v3.1.10 (3.1.10)
 pgfkeys.sty    
pgfkeys.code.tex
 pgfmath.sty    
pgfmath.code.tex
pgffor.code.tex
tikz.code.tex
l3backend-pdftex.def    2023-01-16 L3 backend support: PDF output (pdfTeX)
supp-pdf.mkii
epstopdf-base.sty    2020-01-24 v2.11 Base part for package epstopdf
epstopdf-sys.cfg    2010/07/13 v1.3 Configuration of (r)epstopdf for TeX Live
 ***********

 )</opt/texlive/2022/texmf-dist/fonts/type1/public/amsfonts/cm/cmr10.pfb></opt/
texlive/2022/texmf-dist/fonts/type1/public/amsfonts/cm/cmsy10.pfb>
Output written on test.pdf (1 page, 19707 bytes).
Transcript written on test.log.
```

</details>

**Comment by hmenke** [comment]
Backtrace

```latex
! Missing character: There is no . ("2E) in font nullfont.
<recently read> .
                 
\pgfplotsforeachtodomain@@ ...t \ifnum \pgf@xa >0.
                                                  01pt \def \pgfplotsretval ...
<argument> ...3000000000,...,5.0000000000\relax {}
                                                  \else \ifpgfplotsforeachun...

\pgfplots@for@reset@break ...gffor@continuetrue #1
                                                  \relax \pgffor@continuetru...

\pgfplotsforeachtodomain ...{}}\pgfplots@loc@TMPa 
                                                  \xdef \pgfplots@glob@TMPa ...

\pgfplots@validate@plot@domain@arguments ...esat }
                                                  \ifx \pgfplotsretval \pgfu...

\pgfplots@addplotimpl@expression@set@options ...s}
                                                  \pgfplots@plot@expression@...

\pgfplots@addplotimpl@expression@curly@ ...#1}{#2}
                                                  \ifpgfplots@curplot@threed...
l.12         \addplot[samples=50] {x^2};
```

**Comment by hmenke** [comment]
Already fixed in 9f4e2b14af77fd196d769e98dbc622d68dd8df88

**Comment by leingang** [comment]
Thank you! I'll look forward to the next release.
